src/BaseBlockTracker.ts

```diff
@@ -27,7 +27,7 @@ export class BaseBlockTracker extends SafeEventEmitter {
 
   private _blockResetTimeout?: ReturnType<typeof setTimeout>;
 
-  constructor(opts: BaseBlockTrackerArgs = {}) {
+  constructor(opts: BaseBlockTrackerArgs) {
```

Comment:
Because this class is exported directly, this would be a breaking change. It seems perfectly fine to me to require options be passed in here, but probably best avoided in this PR for the sake of simplicity in avoiding breaking changes.

Reply:
Hmm, actually, maybe we're already in "breaking changes" territory due to the other changes.

Reply:
When removing this, the coverage reporter thinks that this line isn't covered. This is likely because none of the subclasses ever call `super()`. I can get around this for now with an `/* istanbul ignore */` and leave a note.

> Hmm, actually, maybe we're already in "breaking changes" territory due to the other changes.

I guess it's not a big deal to issue a new major release, but I wasn't trying to necessarily break any changes with this commit — is there anything that concerns you?

Reply:
The way to test that line would be to test this class directly. It is expected from this package after all. Though for what reason, I have no idea. I don't know what it would be used for.

The changes to `_start` and `_end` are also breaking. They would break subclasses that didn't adjust those methods accordingly.

Reply:
Perhaps the better option would be to stop exporting this class, and have that be the sole breaking change. Then these "internal" changes actually become internal, and you can remove that uncovered line again.

Reply:
Ah, I didn't know that BaseBlockTracker was being exported. I'll try removing that.

Reply:
Changed in [c93e930](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/c93e930fe381725bb400d3d461bf09d2fd8332c8).